packages/next/build/webpack-config.ts

```diff
@@ -377,14 +379,21 @@ export default async function getBaseWebpackConfig(
   try {
     typeScriptPath = require.resolve('typescript', { paths: [dir] })
   } catch (_) {}
-  const tsConfigPath = path.join(dir, 'tsconfig.json')
+  const { tsconfigPath } = config.typescript
+  const tsConfigName = tsconfigPath ?? 'tsconfig.json'
```

Comment:
Let's set the default value to `tsconfig.json` so we don't need to add all these fallbacks

https://github.com/vercel/next.js/blob/0976a04e8ca4f5a5842ddb26069e51e728a64799/packages/next/server/config-shared.ts#L154-L156